### DISCUSSION TITLE
Create a Vscode Workspace with `manim init`

### DISCUSSION THREAD

**Discussion by naveen521kk** [ManimCommunity/manim#2803] [Suggestions and Proposals]
## Description of the proposed feature
When using `manim init` or `manim new` it would be nice if a [VSCode Workspace](https://code.visualstudio.com/docs/editor/workspaces) is created; where the default python interpreter is set to where manim is installed. The workspace should only be created when VSCode is found to be installed(probably using registry stuff on windows/checking for `code` executable). 

## How can the new feature be used?
Users can activate the Workspace by clicking the workspace file. This should solve many issues from users.
This feature would help in:
- Default python executable is set to where Manim is installed (useful when the user used chocolatey)
- Reduce many question from users on how to use Manim with VSCode/VScode's Jupyter extension.

## Additional comments
There should be something similar for other major IDE's(like PyCharm) also and should be implemented.

**Comment by huguesdevimeux** [comment]
I think there should be a flag instead of automatically creating one. 

For my personal case, I have plenty of Python IDE, and while I have VS Code installed, I don't use it for manim :/

**Comment by naveen521kk** [comment]
> I think there should be a flag instead of automatically creating one.

👍, probably there should be an option in the config file whether to create it by default.

**Comment by naveen521kk** [comment]
I will work on this issue.

**Comment by Darylgolden** [comment]
> 
> 
> I will work on this issue.

Wouldn't this be more suitable of a feature for a VSCode plugin? I don't think such functionality should exist within the core library itself.

**Comment by naveen521kk** [comment]
> Wouldn't this be more suitable of a feature for a VSCode plugin? I don't think such functionality should exist within the core library itself.

Not sure how will a VSCode plugin would work, but IMO it should be fine to have such stuff inside the core library. I think users can benefit a lot; when they have multiple python installations and vscode not able to find manim(especially when people using chocolatey).
